habitat/utils/env_utils.py

```diff
@@ -26,11 +26,11 @@ def make_env_fn(
     Returns:
         env object created according to specification.
     """
-    dataset = make_dataset(
-        config.TASK_CONFIG.DATASET.TYPE, config=config.TASK_CONFIG.DATASET
-    )
+    if "TASK_CONFIG" in config:
```

Comment:
What's the purpose of this? Shouldn't this only be through config, not the TASK_CONFIG as with habitat baselines configurations?

Reply:
The reason is because I want to allow the user to pass either a task config or a baseline config to make the environment. If the user passes a baseline config, this code will look at the task config and use it. Does that make sense ?